src/base/bittorrent/session.cpp

```diff
@@ -3894,56 +3899,94 @@ void Session::startUpTorrents()
         ++resumedTorrentsCount;
     };
 
-    qDebug("Starting up torrents");
+    qDebug("Starting up torrents...");
     qDebug("Queue size: %d", fastresumes.size());
-    // Resume downloads
-    QMap<int, TorrentResumeData> queuedResumeData;
-    int nextQueuePosition = 1;
-    int numOfRemappedFiles = 0;
+
     const QRegularExpression rx(QLatin1String("^([A-Fa-f0-9]{40})\\.fastresume$"));
-    foreach (const QString &fastresumeName, fastresumes) {
-        const QRegularExpressionMatch rxMatch = rx.match(fastresumeName);
-        if (!rxMatch.hasMatch()) continue;
 
-        QString hash = rxMatch.captured(1);
-        QString fastresumePath = resumeDataDir.absoluteFilePath(fastresumeName);
-        QByteArray data;
-        CreateTorrentParams torrentParams;
-        MagnetUri magnetUri;
-        int queuePosition;
-        if (readFile(fastresumePath, data) && loadTorrentResumeData(data, torrentParams, queuePosition, magnetUri)) {
-            if (queuePosition <= nextQueuePosition) {
-                startupTorrent({ hash, magnetUri, torrentParams, data });
-
-                if (queuePosition == nextQueuePosition) {
-                    ++nextQueuePosition;
-                    while (queuedResumeData.contains(nextQueuePosition)) {
-                        startupTorrent(queuedResumeData.take(nextQueuePosition));
+    QFile queueFile {resumeDataDir.absoluteFilePath(QLatin1String {"queue"})};
+    // TODO: Remove the deprecated code below at v4.2
+    // === BEGIN DEPRECATED CODE === //
+    if (!queueFile.exists()) {
```

Comment:
@glassez:  Given the way this "deprecated code" is triggered, how would someone deliberately deliberately switch to the new storage method? 

Would it simply be a matter of unchecking + rechecking `Enable torrent queuing`, or would I have to manually create a blank "queueFile"?

Reply:
No. You just run qBittorrent with this changes merged. It anyway stores the queue file at application exit.